types/services.d.ts

```diff
@@ -213,6 +209,205 @@ declare namespace Realm {
+             * A document got inserted into the collection.
+             */
+            type InsertEvent<T extends Document> = {
+                /** The type of operation which was performed on the document. */
+                operationType: OperationType.Insert;
+                /** The namespace (database and collection) of the document got inserted into. */
+                ns: DocumentNamespace;
+                /** A document that contains the _id of the inserted document. */
+                documentKey: DocumentKey<T["_id"]>;
+                /** The new document created by the operation */
+                fullDocument: T;
+            } & BaseChangeEvent;
+
+            type UpdateEvent<T extends Document> = {
+                /** The type of operation which was performed on the document. */
+                operationType: OperationType.Update;
+                /** The namespace (database and collection) of the updated document. */
+                ns: DocumentNamespace;
+                /** A document that contains the _id of the updated document. */
+                documentKey: DocumentKey<T["_id"]>;
+                /** A document describing the fields that were updated or removed. */
+                updateDescription: UpdateDescription;
+                /**
+                 * For change streams opened with the `fullDocument: updateLookup` option, this will represents the most current majority-committed version of the document modified by the update operation.
+                 */
+                fullDocument?: T;
+            } & BaseChangeEvent;
+
+            /**
+             * A document got replaced in the collection.
+             */
+            type ReplaceEvent<T extends Document> = {
+                /** The type of operation which was performed on the document. */
+                operationType: OperationType.Replace;
+                /** The namespace (database and collection) of the document got replaced within. */
+                ns: DocumentNamespace;
+                /** A document that contains the _id of the replaced document. */
+                documentKey: DocumentKey<T["_id"]>;
+                /** The document after the insert of the replacement document. */
+                fullDocument: T;
+            } & BaseChangeEvent;
+
+            /**
+             * A document got deleted from the collection.
+             */
+            type DeleteEvent<T extends Document> = {
+                /** The type of operation which was performed on the document. */
+                operationType: OperationType.Delete;
+                /** The namespace (database and collection) which the document got deleted from. */
+                ns: DocumentNamespace;
+                /** A document that contains the _id of the deleted document. */
+                documentKey: DocumentKey<T["_id"]>;
+            } & BaseChangeEvent;
+
+            /**
+             * Occurs when a collection is dropped from a database.
```

Comment:
Maybe `Emitted` instead of `Occurs`

Reply:
Technically, they won't be emitted. Since @RedBeard0531 choose to go for a generator function over an event emitter pattern a change events gets "yielded" from the generator.